sw/airborne/modules/computer_vision/video_thread.c

```diff
@@ -80,6 +80,9 @@ static void *video_thread_function(void *data)
 {
   struct video_config_t *vid = (struct video_config_t *)data;
 
+  char print_tag[80];
+  sprintf(print_tag, "video_thread-%s", vid->dev_name);
```

Comment:
Difficult problem. In this case optical flow fits perfectly in the normal image pipeline, but I can understand that algorithms that take longer might want to only have a reference to this image. 

One solution is to add some kind of variable to the image struct (or list) that indicates for each function that uses the image whether this function still needs the image...
At the moment the image gets read from the camera we can take a new piece of memory when the previous image is not fully processed yet.... 

Note that this: 
- might take a lot of memory (potential problem on the ARDrone). 
- is easy to get bugs in (especially with new students who forget to indicate that they are done). 


Maybe a circular image buffer could also work (solves the memory problem), but then your functions don't have control over when the image is re-used again. 

Anyway, this does not matter for this pull request I think.